scripts/dream.py

```diff
@@ -220,10 +222,70 @@ def main_loop(gen, opt, infile):
             prior_variations = opt.with_variations or []
             first_seed       = opt.seed
 
+            if opt.save_progress is not None or opt.show_progress is not None:
+                step_index = 1
+                step_count = 1
+
+            if opt.show_progress is not None:
+                if len(opt.show_progress) < 2:
+                    if len(opt.show_progress) == 0:
+                        opt.show_progress.extend([5, 2])
+                    elif len(opt.show_progress) == 1:
+                        opt.show_progress.append(2)
+                step_count = int(opt.show_progress[0])
+            
+            if opt.save_progress is not None:                                
+                if len(opt.save_progress) < 2:
+                    if len(opt.save_progress) == 0:
+                        opt.save_progress.extend([5, None])
+                    elif len(opt.save_progress) == 1:
+                        opt.save_progress.append(None)
+
+                step_count, progress_video_type = opt.save_progress
+            
+                if progress_video_type is not None:
+                    frames_for_video = []
+
+                if progress_video_type != 'vo':
+                    step_writer = PngWriter(os.path.join(current_outdir, 'intermediates'))
+                                
+            def image_progress(sample, step):
+                nonlocal step_index
+                nonlocal step_count
+
+                if step_count == 0:
+                    step_count = 5                 
+                
+                if step % int(step_count) == 0 and step < opt.steps - 1:
```

Comment:
There needs to be some type checking on the argument you pass to -save_progress or -show_progress. If you pass a non-numeric argument (as I just did with `-save-progress vo` you get a crash.

Reply:
Agreed. I'm out at the moment. Can't work on this for a bit. If you wanna go ahead and add these tweaks in and make a PR here, I don't mind. If not, I'll do it when I'm back home.